src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,78 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] rows = readLines(fromFileName);
+        StringBuilder stringBuilder = createReport(rows);
+        saveReport(toFileName, stringBuilder.toString());
+    }
+
+    private void saveReport(String toFileName, String report) {
+        File file = new File(toFileName);
+        BufferedWriter bufferedWriter = null;
+        try {
+            bufferedWriter = new BufferedWriter(new FileWriter(file));
```

Comment:
try-with-resources
Remember, if you are using classes that implement an AutoCloseable interface, we should use it with try-with-resources.